src/ape/api/providers.py

```diff
@@ -133,6 +134,27 @@ def decode(cls, data: dict) -> "ReceiptAPI":
         ...
 
 
+@abstractdataclass
+class BlockAPI:
+    base_fee_per_gas: Optional[int] = None
```

Comment:
Wonder if gas fee info could be abstracted out into `gas_fee_data: dict` or into a computed property, so that non-EIP1559 blocks can produce the same sorts of data you'd get from EIP1559 blocks

Reply:
`gas_limit` and `gas_used` could be here as well :thinking: